### PR TITLE
GO-3218 Wipe bleve properly

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: pkg/lib/localstore/ftsearch/ftsearch.go [hunk 1]
```diff
@@ -224,8 +230,20 @@ func (f *ftSearch) Has(id string) (exists bool, err error) {
 	return d != nil, nil
 }
 
+func (f *ftSearch) getDocId(id string) ([]string, error) {
```
Comment (deff7) [inline pkg/lib/localstore/ftsearch/ftsearch.go:233]: `getDocIdsForObjectId`? or `collectDocIdsForObjectId`

File: pkg/lib/localstore/ftsearch/ftsearch.go [hunk 2]
```diff
@@ -27,20 +27,22 @@ import (
...
 	ftsDir = "fts"
-	ftsVer = "3"
+	ftsVer = "4"
 
 	fieldTitle        = "Title"
 	fieldText         = "Text"
 	fieldTitleNoTerms = "TitleNoTerms"
 	fieldTextNoTerms  = "TextNoTerms"
 	fieldID           = "Id"
+	fieldDocID        = "DocId"
```
Comment (deff7) [inline pkg/lib/localstore/ftsearch/ftsearch.go:37]: ID -> Id

File: core/indexer/fulltext.go [hunk 3]
```diff
@@ -55,6 +55,7 @@ func (i *indexer) runFullTextIndexer() {
 	docs := make([]ftsearch.SearchDoc, 0, ftBatchLimit)
 	err := i.store.BatchProcessFullTextQueue(ftBatchLimit, func(ids []string) error {
 		for _, id := range ids {
+			_ = i.ftsearch.Delete(id)
```
[the remark this answers is not in the document]
Comment (fat-fellow) [reply core/indexer/fulltext.go:58]: yep, it is for removing the old data

File: pkg/lib/localstore/ftsearch/ftsearch.go [hunk 4]
```diff
@@ -157,8 +159,12 @@ func (f *ftSearch) BatchDelete(ids []string) (err error) {
 			l.Debugf("ft delete done")
 		}
 	}()
+
 	for _, id := range ids {
-		batch.Delete(id)
+		docIds, _ := f.getDocIdsForObjectId(id)
```
Comment (deff7) [inline pkg/lib/localstore/ftsearch/ftsearch.go:164]: error isn't handled

**Review by deff7 — APPROVED** [review]

**Pull request merged**